command/secrets/init/params.go

```diff
@@ -137,41 +134,51 @@ func (ip *initParams) initLocalSecretsManager() error {
 }
 
 func (ip *initParams) initValidatorKey() error {
-	validatorKey, err := helper.InitValidatorKey(ip.secretsManager)
-	if err != nil {
-		return err
+	var err error
+
+	if ip.generatesECDSA {
+		if _, err = helper.InitECDSAValidatorKey(ip.secretsManager); err != nil {
+			return err
+		}
 	}
 
-	ip.validatorPrivateKey = validatorKey
+	if ip.generatesBLS {
+		if _, err = helper.InitBLSValidatorKey(ip.secretsManager); err != nil {
+			return err
+		}
+	}
 
 	return nil
 }
 
 func (ip *initParams) initNetworkingKey() error {
-	networkingKey, err := helper.InitNetworkingPrivateKey(ip.secretsManager)
-	if err != nil {
-		return err
+	if ip.generatesNetwork {
+		if _, err := helper.InitNetworkingPrivateKey(ip.secretsManager); err != nil {
+			return err
+		}
```

Comment:
Why would you **not** want to generate a networking private key?

Reply:
As you see the default value of `ip.generatesNetwork` (default: true), this is for disabling key generation of network key. 
When you add new BLS key, this part throws error because a network key already exists.

Reply:
I finaly got it why we need all of these flags.
Image that you have a running chain with 4 validators, and you want to switch to bls. In your secrets you need to add bls key. You can not use:
`go run main.go secrets init --data-dir test-chain-1 --bls`
because ecdsa an network keys already exists. So you need to use:
`go run main.go secrets init --data-dir test-chain-1 --network=false --ecdsa=false --bls`
to disable generation of network and ecdsa keys and only to generate bls

Am i right @Kourin1996 ?

Reply:
Yes, as explained in regression test document
@Aleksao998

Reply:
@zivkovicmilos ping

Reply:
Didn't we discuss about separating out the BLS key generation in a separate command?

Reply:
@zivkovicmilos Have you added to subtask? I remember we've not decided it. If you feel the need, please create card.

Reply:
Edge-757